ddtrace/tracer/tracer.go

```diff
@@ -254,7 +254,7 @@ func Start(opts ...StartOption) error {
 	cfg.AppVersion = t.config.internalConfig.Version()
 	cfg.Env = t.config.internalConfig.Env()
 	cfg.HTTP = t.config.httpClient
-	cfg.ServiceName = t.config.serviceName
+	cfg.ServiceName = t.config.internalConfig.ServiceName()
```

Comment:
Given that we have gotten rid of the call to `getDDorOtelConfig` above, why don't we have to use ConfigProvider here?